test/makerV2Manager_invest.js

```diff
@@ -1,216 +1,115 @@
-// 2. npx etherlime test test/makerV2Manager_invest.js --skip-compilation --network kovan
-// --- TO TEST ON KOVAN-FORK: ----
-// 2. npm run kovan-fork
-// 3. npx etherlime test test/makerV2Manager_invest.js --skip-compilation --network kovan-fork
-
-const { parseEther } = require("ethers").utils;
+const { AddressZero } = require("ethers").constants;
+const { WAD } = require("../utils/utilities.js");
+const { deployMaker } = require("../utils/defi-deployer");
 const TestManager = require("../utils/test-manager");
-const DeployManager = require("../utils/deploy-manager.js");
-const UniswapFactory = require("../lib/uniswap/UniswapFactory");
-const UniswapExchange = require("../lib/uniswap/UniswapExchange");
-const ScdMcdMigration = require("../build/ScdMcdMigration");
-const Join = require("../build/JoinLike");
-const MakerV2Manager = require("../build/MakerV2Manager");
-const MakerRegistry = require("../build/MakerRegistry");
+const MakerV2Invest = require("../build/TestMakerV2Invest");
 const Wallet = require("../build/BaseWallet");
-const DSToken = require("../build/DSToken");
+const GuardianStorage = require("../build/GuardianStorage");
 
-const DEFAULT_NETWORK = "kovan"; // a bug in kovan-fork makes some tests fail => use kovan
-const SENT_AMOUNT = parseEther("0.00000001");
+const DAI_SENT = WAD.div(100000000);
 
-describe("Test MakerV2 DSR & SAI<>DAI", function () {
-  this.timeout(1000000);
+/* global accounts */
+describe("MakerV2 DSR", function () {
+  this.timeout(100000);
 
-  if (!process.argv.join(" ").includes(__filename.slice(__dirname.length + 1))) {
-    // We do not want to run this file as part of the complete test suite.
-    // This is because this test file can currently only be run on kovan or kovan-fork
-    // (using the --fork option of ganache-cli) whereas other test files are run using ganache
-    return;
-  }
+  const manager = new TestManager();
+  const { deployer } = manager;
+
+  const infrastructure = accounts[0].signer;
+  const owner = accounts[1].signer;
 
   let wallet;
-  let testManager;
   let makerV2;
-  let saiToken;
-  let daiToken;
-  let walletAddress;
-  let owner;
-  let deployer;
+  let sai;
+  let dai;
 
   before(async () => {
-    const idx = process.argv.indexOf("--network");
-    const network = idx > -1 ? process.argv[idx + 1] : DEFAULT_NETWORK;
-    if (!network.includes("kovan")) throw new Error("--network must be 'kovan' or 'kovan-fork'");
-
-    const deployManager = new DeployManager(network);
-    await deployManager.setup();
-    const { configurator } = deployManager;
-    deployer = deployManager.deployer;
-    testManager = new TestManager([...Array(10)].map(() => deployer), network, deployer);
-    owner = deployer.signer;
-    const { config } = configurator;
-
-    const makerRegistry = await deployer.deploy(MakerRegistry);
+    const m = await deployMaker(deployer, infrastructure);
+    [sai, dai] = [m.sai, m.dai];
+    const { migration, pot } = m;
```

Comment:
why not do 
```
[sai, dai, migration, pot] = [m.sai, m.dai, m.migration, m.pot];
```
or 
```
const { sai, dai, migration, pot } = m;
```

Reply:
I like to keep variables as local as they can be. `pot` and `migration` are only used in `before()` so they can be declared locally and destructured with `{}` but `sai` and `dai` are used in multiple functions so they need to be declared out of the scope of these functions and can therefore not be destructured with `{}` (you can only use variable destructuring at the point of declaration).